### PR TITLE
Switch RMSNorm implementation to native PyTorch version

### PR DESCRIPTION
**Pull request by wwwjn**
## Description
1. Switch RMSNorm implementation to native PyTorch version
2. Remove torchtitan local RMSNorm implementation

### Performance metrics 
Test by training llama_8b model after switching to native nn.RMSNorm  (training.seed = 0, deterministic=true):
   - Before the change:  memory: 49.58GiB(52.20%)  tps: 5,233
   - After the change:  memory: 49.66GiB(52.28%)  tps: 5,173

### Numerics 
Test by training llama_8b model after switching to native nn.RMSNorm  (training.seed = 0, deterministic=true): 
- Loss:
  - Before the change: blue line
  - After the change:  orange line
  - The blue line and orange line is not exactly identical. From @tianyu-l 's comment: Before the change, the downcast from FP32 to BF16 happens before the final output * self.weight whereas in pytorch kernel, the downcast happens after the final matmul .
  
Screenshot 2025-03-19 at 9 05 53 PM

### Profiling of nn.RMSNorm
Test by training debug_model after switching to native nn.RMSNorm:
- Before the change: Screenshot 2025-03-19 at 5 02 27 PM

- After the change:Screenshot 2025-03-19 at 9 10 22 PM

- Note: `Aten::to` is explicitly changing the precision

### CODE REVIEW

**Review by lessw2020 — APPROVED** [review]
great to see this.  
lgtm

**Review by fegin — APPROVED** [review]
LGTM, please ensure the integration tests are green before landing.

File: torchtitan/models/norms.py [hunk 1]
```diff
@@ -31,36 +30,6 @@ def build_norm(norm_type: str, dim: int, eps: float = 1e-6):
     elif norm_type == "np_layernorm":
         return nn.LayerNorm(dim, eps=eps, elementwise_affine=False, bias=False)
     elif norm_type == "rmsnorm":
-        return RMSNorm(dim, eps=eps)
+        return nn.RMSNorm(dim, eps=eps, elementwise_affine=False)
```
Comment (tianyu-l) [inline torchtitan/models/norms.py]:
```suggestion
        return nn.RMSNorm(dim, eps=eps)
```

Comment (tianyu-l) [reply torchtitan/models/norms.py]: Here's the kernel implementation of `nn.RMSNorm`
https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/layer_norm.cpp#L264
It seems the two implementations are different, at least in the following aspects:
1. Before the change, the handwritten RMSNorm is actually with `elementwise_affine=True`, which means the affine weights are trainable. We should make it consistent with old behavior.
2. Before the change, the downcast from FP32 to BF16 happens before the final `output * self.weight` whereas in pytorch kernel, the downcast happens after the final matmul (which means it is doing more high-precision computations). I think this should be fine.

Both these two points can be observed in the profiler trace comparison.
cc: @lessw2020 @fegin

**Review by tianyu-l — APPROVED** [review]
Looks great! Congratulations on your first torchtitan PR!

P.S.: Despite the subtle difference between the order of downcast and final matmul, and the tiny differences between loss curves (due to the reason above, or some other reasons e.g. kernel difference) which is hard to make identical, the PR seems very solid.

File: torchtitan/models/norms.py (hunk 1 above)
Comment (fegin) [reply torchtitan/models/norms.py]: Nice catch. Do we know if the handwritten RMSNorm is actually consistent with the llama implementation? My guess it is but just want to confirm.
